### PR TITLE
[FE-17951] D3 palette mappings

### PR DESCRIPTION
**Pull request by cmatzenbach**
See Immerse PR for details

### CODE REVIEW

File: src/charts/pie-chart.js [hunk 1]
```diff
@@ -182,15 +181,9 @@ export default function pieChart(parent, chartGroup) {
...
-    if (_chart.customDomain().length > 0 && chartData) {
-      pieData = chartData.filter(d => _chart.customDomain().includes(d.key0))
-    } else if (chartData) {
-      pieData = chartData
-    }
-
     // if we have data...
-    if (pieData && d3.sum(pieData, _chart.valueAccessor())) {
-      pieData = pie(utils.maybeFormatInfinity(pieData))
+    if (chartData && d3.sum(chartData, _chart.valueAccessor())) {
```
Comment (cmatzenbach) [inline src/charts/pie-chart.js:185]: Reverting this back to how it was before...do not want to filter data based off immerse-set domain

File: src/charts/pie-chart.js [hunk 2]
```diff
@@ -140,13 +146,6 @@ export default function pieChart(parent, chartGroup) {
         })
     })
   })
-
-  override(_chart, "getColor", (data, index) => {
```
Comment (cmatzenbach) [inline src/charts/pie-chart.js:144]: No longer need override since others are handled below

File: src/utils/color-helpers.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    ])
+  )
+  chart.customDomain([...dataMap.keys()])
+  chart.customRange([...dataMap.values()])
+}
+
+export function maybeUpdateAllOthers(chart, data, domain, range) {
+  if (
+    data.map(d => d.data.key0).includes(ALL_OTHERS_LABEL) &&
+    !domain.includes()
```
Comment (johallar) [inline src/utils/color-helpers.js:28]: What's up with this `includes()` check?

File: src/utils/color-helpers.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  ) {
+    domain.push(ALL_OTHERS_LABEL)
+    range.push(ALL_OTHERS_COLOR)
+  } else if (
+    domain.includes(ALL_OTHERS_LABEL) &&
+    !data.map(d => d.data.key0).includes(ALL_OTHERS_LABEL)
```
Comment (johallar) [inline src/utils/color-helpers.js:34]: Data is looped through twice in this function, would be good to extract to variable at the top of the function.

File: src/utils/color-helpers.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  data,
+  dataAccessor,
+  domain,
+  range,
+  wrapKey = false
+) {
+  const dataArr = data.map(dataAccessor)
+  const domainRangeMap = new Map([...domain].map((key, i) => [key, range[i]]))
+  const dataMap = new Map(
+    [...dataArr].map((key, i) => [
```
Comment (johallar) [inline src/utils/color-helpers.js:15]: `[...domain]` and `[...dataArr]` not necessary, `map` returns a new array

File: src/utils/color-helpers.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+      key,
+      domainRangeMap.get(key) ??
+        chart.getColor(wrapKey ? { key0: key } : key, i)
+    ])
+  )
+  chart.customDomain([...dataMap.keys()])
```
Comment (johallar) [inline src/utils/color-helpers.js:21]: I don't think these spread operators are necessary either, `keys` and `values` return new arrays

File: src/utils/color-helpers.js (hunk 3 above)
Comment (cmatzenbach) [reply src/utils/color-helpers.js]: Bad auto find and replace lol

File: src/utils/color-helpers.js (hunk 4 above)
Comment (cmatzenbach) [reply src/utils/color-helpers.js:34]: Good point, extracted both of those to the top of the function

**Review by johallar — APPROVED** [review]

**Pull request merged**